### PR TITLE
Revamp volume configuration, fix issues introduced in 2.4.0

### PR DESCRIPTION
**Pull request by pepov**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | yes
| API breaks?     | yes
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
 - Remove the default `hostPath` volumes for fluentbit, introduced by mistake in 2.4.0 and use `emptyDir` instead.
 - Make the path of the `hostPath` volumes automatically configured if not set
 - Remove the default volume-mount-hack and use `fsGroup` instead
 - Add missing `runAsGroup` for container `securityContext`

### Why?
Because hostPath volumes can cause issues on readonly host filesystems.

### Checklist

- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

File: pkg/resources/fluentd/statefulset.go [hunk 1]
```diff
@@ -48,6 +48,23 @@ func (r *Reconciler) statefulset() (runtime.Object, k8sutil.DesiredState) {
 }
 
 func (r *Reconciler) statefulsetSpec() *appsv1.StatefulSetSpec {
+	initContainers := make([]corev1.Container, 0)
+
+	if !r.Logging.Spec.FluentdSpec.DisablePvc && r.Logging.Spec.FluentdSpec.VolumeMountChmod {
```
Comment (tarokkk) [inline pkg/resources/fluentd/statefulset.go]: Do we need the PVC part for this condition?

File: pkg/sdk/api/v1beta1/common_types.go [hunk 2]
```diff
@@ -37,6 +39,41 @@ type Metrics struct {
...
+//           logging system can coexist in case a hostPath mount is used
+// `name`    will be the name of the volume and the lowest level directory in case a hostPath mount is used
+func (storage *KubernetesStorage) GetVolume(logging, name string) corev1.Volume {
+	const (
+		hostPath = "/opt/logging-operator/%s/%s"
+	)
+	volume := corev1.Volume{
+		Name: name,
+	}
+	if storage != nil {
```
Comment (tarokkk) [inline pkg/sdk/api/v1beta1/common_types.go]: We should do this check before invoking the function.

File: pkg/resources/fluentd/statefulset.go (hunk 1 above)
Comment (pepov) [reply pkg/resources/fluentd/statefulset.go]: not really, I've removed it

File: pkg/sdk/api/v1beta1/common_types.go (hunk 2 above)
Comment (pepov) [reply pkg/sdk/api/v1beta1/common_types.go]: modified to be a value receiver

**Pull request merged**